subsys/bluetooth/controller/ll_sw/ull_central_iso.c

```diff
@@ -60,75 +120,513 @@ uint8_t ll_cig_parameters_test_open(uint8_t cig_id,
+		 *  ------------------------------------------------------------------------
+		 *  10 ms          10 ms          40        40        100%
+		 *  10 ms          12.5 ms        40        50         25%
+		 */
+		iso_interval_us = cig->c_sdu_interval;
+		cig->iso_interval = ceiling_fraction(iso_interval_us, ISO_INT_UNIT_US);
+	} else {
+		iso_interval_us = cig->iso_interval * ISO_INT_UNIT_US;
+	}
+
+	ull_hdr_init(&cig->ull);
+	lll_hdr_init(&cig->lll, cig);
+
+	max_se_length = 0;
+	cis_count = cig->cis_count;
+
+	/* 1) Acquire CIS instances and initialize instance data.
+	 * 2) Calculate SE_Length for each CIS and store the largest
+	 * 3) Calculate BN
+	 * 4) Calculate total number of subevents needed to transfer payloads
+	 *
+	 *                 Sequential                Interleaved
+	 * CIS0            ___█_█_█_____________█_   ___█___█___█_________█_
+	 * CIS1            _________█_█_█_________   _____█___█___█_________
+	 * CIS_Sub_Interval  |.|                       |...|
+	 * CIG_Sync_Delay    |............|            |............|
+	 * CIS_Sync_Delay 0  |............|            |............|
+	 * CIS_Sync_Delay 1        |......|              |..........|
+	 * ISO_Interval      |.................|..     |.................|..
+	 */
+	for (uint8_t i = 0; i < cis_count; i++) {
+		uint32_t mpt_c;
+		uint32_t mpt_p;
+
+		const bool tx = cig->c_sdu_interval > 0;
+		const bool rx = cig->p_sdu_interval > 0;
+
+		/* Acquire new CIS */
+		cis = ll_conn_iso_stream_acquire();
+		if (cis == NULL) {
+			/* No space for new CIS */
+			return BT_HCI_ERR_INSUFFICIENT_RESOURCES;
+		}
+
+		/* Transfer parameters from update cache */
+		memcpy(cis, &ll_iso_setup.stream[i], sizeof(struct ll_conn_iso_stream));
+		cis->group = cig;
+
+		cis->lll.handle = ll_conn_iso_stream_handle_get(cis);
+
+		if (cig->central.test) {
+			cis->lll.tx.flush_timeout = ll_iso_setup.c_ft;
+			cis->lll.rx.flush_timeout = ll_iso_setup.p_ft;
+		} else {
+			LL_ASSERT(iso_interval_us >= cig->c_sdu_interval);
+
+			if (cis->framed) {
+				cis->lll.tx.max_octets =
+					ceiling_fraction(cis->c_max_sdu * cig->c_sdu_interval,
+							 iso_interval_us);
+				cis->lll.rx.max_octets =
+					ceiling_fraction(cis->p_max_sdu * cig->c_sdu_interval,
+							 iso_interval_us);
+			} else {
+				/* For unframed, ISO_Interval must be N x SDU_Interval */
+				LL_ASSERT(iso_interval_us % cig->c_sdu_interval == 0);
+
+				/* Use Max_PDU = MIN(<buffer_size>, Max_SDU) */
+				cis->lll.tx.max_octets = MIN(CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE,
+							 cis->c_max_sdu);
+				cis->lll.rx.max_octets = MIN(251, cis->p_max_sdu);
+			}
+
+			/* BN >= Max_SDU/Max_PDU * ISO_Interval/SDU_Interval */
+			cis->lll.tx.burst_number = tx ?
+				ceiling_fraction(cis->c_max_sdu * iso_interval_us,
+						 cis->lll.tx.max_octets * cig->c_sdu_interval) : 0;
+			cis->lll.rx.burst_number = rx ?
+				ceiling_fraction(cis->p_max_sdu * iso_interval_us,
+						 cis->lll.rx.max_octets * cig->p_sdu_interval) : 0;
```

Comment:
This fixes the floating point exception in central_iso sample. Now I have the central_iso and peripheral_iso  sucessfully scheduling the ticker callbacks (have to use `.rtn = 0U,`  though in the sample to avoid assertion check on the latency value). 

```suggestion
			cis->lll.tx.burst_number = (tx && cis->lll.tx.max_octets) ?
				ceiling_fraction(cis->c_max_sdu * iso_interval_us,
						 cis->lll.tx.max_octets * cig->c_sdu_interval) : 0;
			cis->lll.rx.burst_number = (rx && cis->lll.rx.max_octets)?
				ceiling_fraction(cis->p_max_sdu * iso_interval_us,
						 cis->lll.rx.max_octets * cig->p_sdu_interval) : 0;
```

Reply:
I think the fix should be different. If `max_octets `is 0, `tx` should be false.

Reply:
As you may have guessed, I have some problems running CI tests locally. Tomorrow I will try to get that up and running again.

Reply:
Fixed this as I think it should be. Please check if it the exception is gone. Thanks.